lib/src/client.dart

```diff
@@ -180,7 +180,7 @@ class NyxxRest with ManagerMixin implements Nyxx {
   }
 }
 
-class NyxxOAuth2 with ManagerMixin implements NyxxRest {
+class NyxxOAuth2 with ManagerMixin, EventRestMixin implements NyxxRest {
```

Comment:
The above comment also solves the issue here as OAuth2 clients _cannot_ receive interaction events (and also cannot connect to the gateway, which is why NyxxOAuth2 doesn't implement NyxxGateway).